inc/sainat.h

```diff
@@ -0,0 +1,476 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    sainat.h
+ *
+ * @brief   This module defines SAI NAT (Network Address Translation) spec
+ */
+
+#if !defined (__SAINAT_H_)
+#define __SAINAT_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAINAT SAI - Network Address Translation (NAT) specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief NAT Entry Attributes
+ */
+typedef enum _sai_nat_entry_attr_t
+{
+
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_NAT_ENTRY_ATTR_START,
+
+    /**
+     * @brief NAT IP address translated from
+     *
+     * @type sai_ip_address_t
```

Comment:
During the call it was stated that the from and to IP addresses could be masked. Looking in saitypes.h, I don't see a mask on sai_ip_address_t. Did you mean to use sai_ip_prefix_t instead?

Reply:
Stale Changes